kernel/src/schema.rs

```diff
@@ -647,61 +647,73 @@ impl Display for DataType {
 /// The provided `recurse_into_xxx` methods encapsulate the boilerplate work of recursing into the
 /// child schema elements of each schema element. Implementations can call these as needed but will
 /// generally not need to override them.
-pub trait SchemaTransform {
+pub trait SchemaTransform<'a> {
```

Comment:
does sharing the lifetime across methods make sense here? put another way: should the lifetimes of the types we are transforming all be the same? I suppose the answer is yes since each method is invokes as we are transforming some type and possibly recursing which would imply the same lifetime requirements?

Reply:
This was an observation by @nicklan that allows the data skipping PR to not clone string field names it stores references to -- everything reachable through the entry point must survive at least as long as the entry point, so we can define a single lifetime for the entire traversal that visitors can take advantage of.

Reply:
okay yep makes sense! thank you!